src/AggregateFunctions/AggregateFunctionIf.cpp

```diff
@@ -136,28 +126,36 @@ class AggregateFunctionIfNullUnary final
         const IColumn * columns_param[] = {&column->getNestedColumn()};
 
         const IColumn * filter_column = columns[num_arguments - 1];
-        if (const ColumnNullable * nullable_column = typeid_cast<const ColumnNullable *>(filter_column))
-            filter_column = nullable_column->getNestedColumnPtr().get();
-        if constexpr (result_is_nullable)
+
+        const UInt8 * filter_values = nullptr;
+        const UInt8 * filter_null_map = nullptr;
+
+        if (filter_is_nullable)
         {
-            /// We need to check if there is work to do as otherwise setting the flag would be a mistake,
```

Comment:
I did not understand it, something strange.

Reply:
I think this was trying to fix the bug (or a similar one) in a different place. I needed to check if the filter had any values to check because otherwise a) there is no work to do and b) the result, for nullable types, should be NULL. The problem came from using `null_map` and not `final_nulls` to do the check though.